index.d.ts needs DOM in lib to work

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
It doesn't compile if the tsconfig.json doesn't include `DOM` in `lib` option. I am not going to run Mongoose on a browser, so I should not be required to include DOM libraries in the environment.
```
node_modules/mongoose/index.d.ts:191:9 - error TS2304: Cannot find name 'WritableStream'.

191       | WritableStream
            ~~~~~~~~~~~~~~


Found 1 error.
```
`git blame` said `WritableStream` came from https://github.com/Automattic/mongoose/pull/10471.

**If the current behavior is a bug, please provide the steps to reproduce.**
Compile this project.

package.json
```json
{
  "name": "test",
  "version": "1.0.0",
  "license": "MIT",
  "dependencies": {
    "mongoose": "^5.13.4"
  },
  "devDependencies": {
    "@types/node": "^14.17.6",
    "typescript": "~4.3.5"
  }
}
```

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "ES2019",
    "module": "commonjs",
    "lib": ["ES2019"],
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": false,
    "forceConsistentCasingInFileNames": true
  }
}
```

index.ts
```
import mongoose from 'mongoose';
```

**What is the expected behavior?**
It should compile without DOM.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node.js 14.17.0
Mongoose 5.13.4
MongoDB 4.4.7

I have this issue too. Version 5.13.4 is broken I get the same error. Please advise.
the error

node_modules/mongoose/index.d.ts:191:9 - error TS2304: Cannot find name 'WritableStream'.

191       | WritableStream

I have this issue too, noticed it while trying out 6.0, project's used `@types/node` version is `~12.12.6` (project's lowest supported nodejs version) (not having tsconfig `compilerOptions.lib.dom` enabled)

it can currently (if wanted) easily reproduced by cloning [this repo](https://github.com/typegoose/typegoose) and checking out the branch `mongoose6.0` and then running `yarn install` and `yarn run build` and the first tsc error that comes up should be about this issue

or an easier light-weight repro:
- clone [this repo](https://github.com/typegoose/typegoose-testing)
- checkout `reproMongoose10504`
- run `yarn install`
- run `yarn run run` (even without having a mongodb instance up)

full tsc error:
```
node_modules/mongoose/index.d.ts:191:9 - error TS2304: Cannot find name 'WritableStream'.

191       | WritableStream
            ~~~~~~~~~~~~~~


Found 1 error.
```

> I'm unable to repro this, can you please confirm what version of `@types/node` you have installed? 

Yes.
```
$ find node_modules -name node
node_modules/mongoose/node_modules/@types/node
node_modules/@types/node
node_modules/@types/mongodb/node_modules/@types/node
node_modules/@types/bson/node_modules/@types/node
```
package.json has `"@types/node": "^14.17.6"`, and it matches the installed version.
```
$ jq .version < node_modules/@types/node/package.json 
"14.17.6"
$ jq .version < node_modules/mongoose/node_modules/@types/node/package.json 
"15.14.4"
$ jq .version < node_modules/@types/mongodb/node_modules/@types/node/package.json 
"16.4.7"
$ jq .version < node_modules/@types/bson/node_modules/@types/node/package.json 
"16.4.7"
```
Reproduce the bug with the files I provided in the description:
```
$ node --version
v14.17.0
$ npm --version
6.14.13
$ ls -l
total 12
-rw-r--r--. 1 lantw44 lantw44  33 Jul 30 11:15 index.ts
-rw-r--r--. 1 lantw44 lantw44 200 Jul 30 11:14 package.json
-rw-r--r--. 1 lantw44 lantw44 225 Jul 30 11:15 tsconfig.json
$ npm install
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN test@1.0.0 No description
npm WARN test@1.0.0 No repository field.

added 38 packages from 96 contributors and audited 38 packages in 1.491s

2 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

$ npx tsc
node_modules/mongoose/index.d.ts:191:9 - error TS2304: Cannot find name 'WritableStream'.

191       | WritableStream
            ~~~~~~~~~~~~~~


Found 1 error.

(npx tsc exited with status 2)
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -188,7 +188,7 @@ declare module 'mongoose' {
     debug?:
       | boolean
       | { color?: boolean; shell?: boolean }
-      | WritableStream
+      | stream.Writable
       | ((collectionName: string, methodName: string, ...methodArgs: any[]) => void);
 
     /** If set, attaches [maxTimeMS](https://docs.mongodb.com/manual/reference/operator/meta/maxTimeMS/) to every query */

